lib/react.dart

```diff
@@ -20,7 +20,12 @@ export 'package:react/react_client/react_interop.dart' show forwardRef, createRe
 typedef Error PropValidator<TProps>(
     TProps props, String propName, String componentName, String location, String propFullName);
 
+/// A React component declared using a function that takes in [props] and returns rendered output.
+///
+/// See <https://facebook.github.io/react/docs/components-and-props.html#functional-and-class-components>.
+typedef DartFunctionComponent = dynamic Function(Map props);
```

Comment:
I think I already asked this - but its not notated in the comment anywhere... what is/are the expected return type(s)? Is `dynamic` the best we can do for the return signature?

Reply:
Yeah this is similar to the return of render, it can technically be a number of different things...
Typescripts render / return type is `ReactNode`:

```typescript

    type ReactText = string | number;
    type ReactChild = ReactElement | ReactText;

    interface ReactNodeArray extends Array<ReactNode> {}
    type ReactFragment = {} | ReactNodeArray;
    type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```

so something like:
```
/* union ReactReturn = ReactElement | List<ReactReturn> | String | num | bool | Null; */

Reply:
cool